src/storage/query/QueryBoundProcessor.cpp

```diff
@@ -86,6 +86,14 @@ kvstore::ResultCode QueryBoundProcessor::processVertex(PartitionID partId, Verte
             }
         }
         vResp.set_tag_data(std::move(td));
+    } else if (tagContexts_.empty() && onlyVertexProps_) {
```

Comment:
Implement the "collect all props for one vertex"  logic inside QueryVertexPropsProcessor